lib/pbench/server/api/resources/upload_api.py

```diff
@@ -70,11 +70,12 @@ def put(self, filename: str):
 
         try:
             try:
+                user_id = Auth.token_auth.current_user().id
                 username = Auth.token_auth.current_user().username
             except Exception:
                 username = None
                 raise CleanupTime(
-                    HTTPStatus.INTERNAL_SERVER_ERROR, "Error verifying username"
+                    HTTPStatus.INTERNAL_SERVER_ERROR, "Error verifying user_id"
                 )
```

Comment:
This is a bit silly, come to think of it. If `Auth.token_auth.current_user()` is `None` then the `PUT` isn't authenticated and we should fail with `HTTPStatus.UNAUTHORIZED`, not an "error validating username". And, in any case, the error message is uselessly misleading. (We could report an internal server error if `Auth.token_auth.current_user()` returns something other than a `User` or `None`, but that seems unnecessarily complicated.)

Reply:
I totally agree, but does it make sense to create a separate ticket and put a PR against it for this change, or should I address it here? It isn't a big change but maybe not related to this work and may warrant its own PR for the improvement to upload api.

Reply:
You'll have to touch this code again when you update `Auth.token_auth` for Keycloak, and presumably that will leverage the hypothetical new "id -> name" method. You can fix it then, I suppose; though if you touch this again for any reason at least improving the message would be nice. (Though if you want to return `UNAUTHORIZED` I'd be happy.)